composer/datasets/streaming/download.py

```diff
@@ -40,7 +40,7 @@ def download_from_s3(remote: str, local: str, timeout: float) -> None:
     s3 = boto3.client('s3', config=config)
     try:
         s3.download_file(obj.netloc, obj.path.lstrip('/'), local)
-    except DataNotFoundError as e:
+    except ClientError as e:
```

Comment:
For detect not found errors, see https://github.com/mosaicml/composer/blob/16622c9457874f4f3811287fe933b0ec58107fc7/composer/utils/object_store/s3_object_store.py#L21

`ClinetError` is too general

Reply:
```
pytest -s -m remote "tests/datasets/test_streaming_remote.py::test_streaming_remote_dataset[val-ade20k]"       

...

E           RuntimeError: Failed to download s3://mosaicml-internal-dataset-ade20k/mds/1/val/compression.txt -> /private/var/folders/4c/5krf_kq12694r_84_8n0ky9c0000gn/T/pytest-of-abhinav/pytest-112/test_streaming_remote_dataset_0/val/compression.txt. Got errors:
E           [ClientError('An error occurred (404) when calling the HeadObject operation: Not Found'), ClientError('An error occurred (404) when calling the HeadObject operation: Not Found'), ClientError('An error occurred (404) when calling the HeadObject operation: Not Found')]

/Users/abhinav/Documents/mosaicml/abhi-composer/composer/datasets/streaming/download.py:190: RuntimeError
```